### PR TITLE
Write FileUploads to a directory under MEDIA_ROOT (/uploads/)

### LINKED ISSUE #18871: Move FileUpload storage to `shared_storage/uploads/fileuploads`
`FileUpload` files are stored in a `temp` directory under `files/` (where permanent add-ons are stored):

https://github.com/mozilla/addons-server/blob/1ac05a5454f04b0985d1b9b540ca3af601882276/src/olympia/files/models.py#L514-L516

This is a bit inconsistent with everything else: `files/` should only really contain permanent xpis (i.e. `File`). We already have a `/shared_storage/tmp/` path for really temporary stuff, and a bunch of other uploads are stored under `shared_storage/uploads/<directory>/`.

Since `FileUpload`s are short-lived but not quite as short-lived as other temporary files that are discarded right after uploads, I think we should use `shared_storage/uploads/fileuploads` instead.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/ADDSRV-80)

### PR DESCRIPTION
**Pull request by diox**
Fixes #18871

### CODE REVIEW

File: src/olympia/files/models.py [hunk 1]
```diff
@@ -508,7 +508,7 @@ def add_file(self, chunks, filename, size):
         if ext in amo.VALID_ADDON_FILE_EXTENSIONS:
             ext = '.xpi'
         self.path = os.path.join(
-            user_media_path('addons'), 'temp', f'{uuid.uuid4().hex}{ext}'
+            settings.MEDIA_ROOT, 'fileuploads', f'{uuid.uuid4().hex}{ext}'
```
Comment (eviljeff) [inline src/olympia/files/models.py:511]:
```suggestion
            user_media_path('fileuploads'), f'{uuid.uuid4().hex}{ext}'
```

**Review by eviljeff — APPROVED** [review]

File: src/olympia/files/models.py (hunk 1 above)
Comment (diox) [reply src/olympia/files/models.py:511]: mmm yes, but... I was kinda thinking about removing that utility function entirely. I find it more confusing that helpful...

Comment (eviljeff) [reply src/olympia/files/models.py:511]: it _is_ a bit "magic"

Comment (diox) [reply src/olympia/files/models.py:511]: https://github.com/mozilla/addons-server/issues/18999

**Pull request merged**